### PR TITLE
Create signTxMetamask tests

### PR DESCRIPTION
**Pull request by AlexChetverov**

### CODE REVIEW

File: utils/metamask.ts [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    const unsub = await api.rpc.chain.subscribeNewHeads(
+      async (header): Promise<void> => {
+        if (counter <= 0) {
+          unsub();
+          reject("Timeout");
+        }
+        counter--;
+        const currentBlock = await api.rpc.chain.getBlock(header.hash);
+        const extrinsics = currentBlock.block.extrinsics;
+        extrinsicResult = extrinsics.filter(
```
Comment (goncer) [inline utils/metamask.ts:175]: add here another 
//TODO: Find some extra determinist way to filter the extrinsic signed by the user.
:pray:

File: utils/metamask.ts [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    nonce: signingInfo.nonce,
+  });
+}
+
+export async function signTxMetamask(
+  tx: any,
+  ethAddress: string,
+  ethPrivateKey: string,
+): Promise<MangataGenericEvent[]> {
+  const api = await ApiPromise.create({
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply utils/metamask.ts:72]: Done

File: utils/metamask.ts (hunk 1 above)
Comment (AlexChetverov) [reply utils/metamask.ts:175]: Done

File: test/parallel/metamask.transfer.test.ts [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+  });
+
+  test("Swap liquidity", async () => {
+    await Sudo.batchAsSudoFinalized(Assets.mintNative(testEthUser.pdAccount));
+    testEthUser.pdAccount.addAsset(MGA_ASSET_ID);
+
+    await testPdUser.refreshAmounts(AssetWallet.BEFORE);
+    await testEthUser.pdAccount.refreshAmounts(AssetWallet.BEFORE);
+
+    const tx = api.tx.tokens.transfer(testPdUser.keyRingPair.address, 0, 1000);
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/parallel/metamask.transfer.test.ts:116]: You're right, fixed

File: utils/metamask.ts [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+    data: data,
+    // @ts-ignore
+    version: "V4",
+  });
+  console.log("Ok, signed typed data ");
+  console.log("SIGNATURE = " + msg_sig);
+  const created_signature = api.createType("MultiSignature", {
+    Eth: hexToU8a(msg_sig),
+  });
+  console.log(tx_payload);
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply utils/metamask.ts:156]: Done

**Review by goncer — APPROVED** [review]

**Pull request merged**